libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt

```diff
@@ -66,10 +65,6 @@ class MapboxDirectionsSession(
         adjustedRouteOptions: RouteOptions,
         routesRequestCallback: RoutesRequestCallback
     ) {
-        if (routes.isEmpty()) {
-            routesRequestCallback.onRoutesRequestCanceled(adjustedRouteOptions)
-            return
-        }
```

Comment:
Do we want to remove this? I believe `requestFasterRoute` could be called (scheduled) while having empty routes (faster route is handled separately in `MapboxNavigation`). Do we want to notify back users through `onRoutesRequestCanceled` as we were doing so clients could stop getting faster route updates? cc @abhishek1508

Reply:
Yeah I don't think removing it is a good idea. I think Oleg had some concerns about it though. I don't have the chat anymore so I don't remember.  cc @Guardiola31337

Reply:
I think the problem is that `requestFasterRoute` could be called in some situations because looking at `MapboxDirectionsSession`

```kotlin
if (routes.isNotEmpty()) {
    this.routeOptions = routes[0].routeOptions()
}
```
we're not resetting the route options and the faster route can be done 😬 because the only thing that we're checking is the route options and the route progress 👀 

```kotlin
fun getRouteOptions(directionsSession: DirectionsSession, tripSession: TripSession, location: Location): RouteOptions? {
    val routeOptions: RouteOptions = directionsSession.getRouteOptions() ?: return null
    val routeProgress: RouteProgress = tripSession.getRouteProgress() ?: return null
    // ...
}
```
but not if there’s a route or not. Does that make sense? cc'ing @kmadsen as you've recently worked on some refactoring around _Faster route_ (`AdjustedRouteOptionsProvider`)

Reply:
It's unclear why faster route would be part of this fix. Faster route creates a dialog that proposes a faster route, assumption is you're on a route. Is the fix to show a dialog for faster route?

And yeah this may have undefined behavior right? What would the [remainingLegDuration](https://github.com/mapbox/mapbox-navigation-android/blob/4fc8f171b3ece8ecabad3d02383b19707098fa5a/libnavigator/src/main/java/com/mapbox/navigation/navigator/MapboxNativeNavigatorImpl.kt#L180) be if there's no Leg? Would have to check with nav native or test it.

Reply:
🆗 I think we should bring this piece of code back for now but we should definitely look at the scenario in which the client transitions from _Active Guidance_ to _Free Drive_ and 👀 how _Faster route_ logic is handled in this situation. As mentioned above it doesn't make any sense to keep trying to request faster routes if you don't have a route at all which may lead to unexpected behavior and errors. Sounds good @abhishek1508 @kmadsen? Will cut a follow up ticket to investigate / fix the issue discussed.

Reply:
Follow up ticket 👉 https://github.com/mapbox/mapbox-navigation-android/issues/2710